airflow/settings.py

```diff
@@ -51,6 +52,36 @@
 log.info("Configured default timezone %s" % TIMEZONE)
 
 
+class RetryingQuery(Query):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        try:
+            self.max_tries = conf.getint('core', 'SQL_ALCHEMY_STATEMENT_MAX_RETRIES')
```

Comment:
Tries != Retries

Reply:
but then retries = total attempts - 1

i struggled with this too.

Reply:
Yes, and that seems normal given how retries are typically configured

Reply:
I made this `max_tries` everywhere. Setting it to <= 1 is equivalent to the original behavior.

Reply:
More comments further down... "retries" (and not "tries") makes more sense as an API.